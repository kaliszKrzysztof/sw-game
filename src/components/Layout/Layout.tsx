import React from 'react';
import { makeStyles, createStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() =>
  createStyles({
    root: {
      minHeight: '100vh',
      paddingBottom: 180,
      position: 'relative',
    },
  }),
);

interface LayoutProps {
  children: React.ReactNode;
}

const Layout: React.FC<LayoutProps> = ({ children }) => {
  const classes = useStyles();
  return <div className={classes.root}>{children}</div>;
};

export default Layout;
