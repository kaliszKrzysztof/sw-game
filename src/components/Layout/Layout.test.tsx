import React from 'react';
import { render } from '@testing-library/react';
import Layout from './Layout';

describe('<Layout />', () => {
  it('should render', () => {
    render(
      <Layout>
        <span>Layout</span>
      </Layout>,
    );
  });
  it('should match snapshot', () => {
    const { container } = render(
      <Layout>
        <span>Layout</span>
      </Layout>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render provided children', () => {
    const { getByTestId } = render(
      <Layout>
        <span data-testid="children">Layout</span>
      </Layout>,
    );
    expect(getByTestId('children')).toBeInTheDocument();
  });
});
