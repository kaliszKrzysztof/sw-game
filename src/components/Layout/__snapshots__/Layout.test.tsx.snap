// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-2"
  >
    <span>
      Layout
    </span>
  </div>
</div>
`;
